ext4: sync up block group descriptor with e2fsprogs.

From: Coly Li <coyli@suse.de>

This patch extends bg_itable_unused of ext4 group descriptor
from 16bit into 32bit. In order to add bg_itable_unused_hi into
struct ext4_group_desc, some extra fields which are already introduced into
e2fsprogs are also added in for consistency.

Signed-off-by: Coly Li <coyli@suse.de>
Cc: Andreas Dilger <adilger@clusterfs.com>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>
---

 include/linux/ext4_fs.h |    5 +++++
 1 file changed, 5 insertions(+)


Index: linux-2.6.24-rc1/include/linux/ext4_fs.h
===================================================================
--- linux-2.6.24-rc1.orig/include/linux/ext4_fs.h	2007-10-24 11:40:15.000000000 -0700
+++ linux-2.6.24-rc1/include/linux/ext4_fs.h	2007-10-24 11:40:21.000000000 -0700
@@ -118,6 +118,11 @@ struct ext4_group_desc
 	__le32	bg_block_bitmap_hi;	/* Blocks bitmap block MSB */
 	__le32	bg_inode_bitmap_hi;	/* Inodes bitmap block MSB */
 	__le32	bg_inode_table_hi;	/* Inodes table block MSB */
+	__le16	bg_free_blocks_count_hi;/* Free blocks count MSB */
+	__le16	bg_free_inodes_count_hi;/* Free inodes count MSB */
+	__le16	bg_used_dirs_count_hi;	/* Directories count MSB */
+	__le16	bg_itable_unused_hi;	/* Unused inodes count MSB */
+	__u32	bg_reserved2[3];
 };
 
 #define EXT4_BG_INODE_UNINIT	0x0001 /* Inode table/bitmap not in use */
